mypy failure seems unrelated:

```
Tools/clinic/clinic.py:2206: error: Item "None" of "Clinic | None" has no
attribute "limited_capi"  [union-attr]
                if not clinic.limited_capi:
                       ^~~~~~~~~~~~~~~~~~~
Found 1 error in 1 file (checked 2 source files)
```

--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -2174,7 +2174,7 @@ class BlockPrinter:
             *,
             core_includes: bool = False,
             limited_capi: bool = False,
-            header_includes: dict[str, str] | None = None,
+            header_includes: dict[str, str],
     ) -> None:
         input = block.input
         output = block.output
@@ -2213,11 +2213,10 @@ class BlockPrinter:
 
                 """)
 
-            if header_includes is not None:
-                # Emit optional "#include" directives for C headers
-                for include, reason in sorted(header_includes.items()):
-                    line = f'#include "{include}"'.ljust(35) + f'// {reason}\n'
-                    output += line
+            # Emit optional "#include" directives for C headers
+            for include, reason in sorted(header_includes.items()):
+                line = f'#include "{include}"'.ljust(35) + f'// {reason}\n'
+                output += line
 
         input = ''.join(block.input)
         output += ''.join(block.output)

--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -2173,6 +2173,7 @@ class BlockPrinter:
             block: Block,
             *,
             core_includes: bool = False,
+            limited_capi: bool = False,
             header_includes: dict[str, str] | None = None,
     ) -> None:
         input = block.input
@@ -2203,7 +2204,7 @@ class BlockPrinter:
 
         output = ''
         if core_includes:
-            if not clinic.limited_capi:
+            if not limited_capi:
                 output += textwrap.dedent("""
                     #if defined(Py_BUILD_CORE) && !defined(Py_BUILD_CORE_MODULE)
                     #  include "pycore_gc.h"            // PyGC_Head

--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -2173,7 +2173,7 @@ class BlockPrinter:
             block: Block,
             *,
             core_includes: bool = False,
-            limited_capi: bool = False,
+            limited_capi: bool,
             header_includes: dict[str, str],
     ) -> None:
         input = block.input
@@ -2530,7 +2530,9 @@ impl_definition block
                     self.parsers[dsl_name] = parsers[dsl_name](self)
                 parser = self.parsers[dsl_name]
                 parser.parse(block)
-            printer.print_block(block, header_includes=self.includes)
+            printer.print_block(block,
+                                limited_capi=self.limited_capi,
+                                header_includes=self.includes)
 
         # these are destinations not buffers
         for name, destination in self.destinations.items():
@@ -2545,7 +2547,9 @@ impl_definition block
                     block.input = "dump " + name + "\n"
                     warn("Destination buffer " + repr(name) + " not empty at end of file, emptying.")
                     printer.write("\n")
-                    printer.print_block(block, header_includes=self.includes)
+                    printer.print_block(block,
+                                        limited_capi=self.limited_capi,
+                                        header_includes=self.includes)
                     continue
 
                 if destination.type == 'file':
@@ -2570,7 +2574,10 @@ impl_definition block
 
                     block.input = 'preserve\n'
                     printer_2 = BlockPrinter(self.language)
-                    printer_2.print_block(block, core_includes=True, header_includes=self.includes)
+                    printer_2.print_block(block,
+                                          core_includes=True,
+                                          limited_capi=self.limited_capi,
+                                          header_includes=self.includes)
                     write_file(destination.filename, printer_2.f.getvalue())
                     continue
 

